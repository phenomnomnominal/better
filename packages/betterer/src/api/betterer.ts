import type { BettererSuiteSummary } from '../suite/index.js';
import type { BettererOptions } from './types.js';

import { BettererRunnerΩ } from '../runner/index.js';
import { merge } from './merge.js';
import { results } from './results.js';
import { runner } from './runner.js';
import { watch } from './watch.js';

/**
 * @public Run **Betterer** with the given options.
 *
 * @example
 * ```typescript
 * import { betterer } from '@betterer/betterer';
 *
 * const suiteSummary = await betterer(options);
 * ```
 *
 * @param options - Options for running **Betterer**.
 * @throws {@link @betterer/errors#BettererError | `BettererError` }
 * Will throw if something goes wrong while running **Betterer**.
 */
export const betterer = async function betterer(options: BettererOptions = {}): Promise<BettererSuiteSummary> {
  const runner = await BettererRunnerΩ.create(options);
  return await runner.run();
};

betterer.merge = merge;
betterer.results = results;
betterer.runner = runner;
betterer.watch = watch;
