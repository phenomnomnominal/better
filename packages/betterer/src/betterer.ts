import { debug } from '@phenomnomnominal/debug';

import { BettererBaseConfigPartial, BettererStartConfigPartial, BettererWatchConfigPartial } from './config';
import { BettererSummary } from './context';
import { BettererRunner, BettererRunnerΩ, BettererWatcherΩ } from './runner';

export async function betterer(partialConfig: BettererStartConfigPartial = {}): Promise<BettererSummary> {
  initDebug();
  const runner = new BettererRunnerΩ();
  await runner.start(partialConfig);
  await runner.queue();
  return runner.stop();
}

export async function runner(partialConfig?: BettererBaseConfigPartial): Promise<BettererRunner> {
  initDebug();
  const runner = new BettererRunnerΩ();
  await runner.start(partialConfig);
  return runner;
}
betterer.runner = runner;

export async function watch(partialConfig: BettererWatchConfigPartial = {}): Promise<BettererRunner> {
  initDebug();
  partialConfig.watch = true;
  const runner = new BettererRunnerΩ();
  const context = await runner.start(partialConfig);
  const watcher = new BettererWatcherΩ(context, runner);
  await watcher.setup();
  return watcher;
}
betterer.watch = watch;

function initDebug(): void {
  const enabled = !!process.env.BETTERER_DEBUG;
  if (enabled) {
    debug({
      header: 'betterer',
      include: [/@betterer\//],
      ignore: [require.resolve('./utils')],
      enabled,
      time: !!process.env.BETTERER_DEBUG_TIME,
      values: !!process.env.BETTERER_DEBUG_VALUES,
      logPath: process.env.BETTERER_DEBUG_LOG
    });
  }
}
