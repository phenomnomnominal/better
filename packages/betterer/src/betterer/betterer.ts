import logDiff from 'jest-diff';

import { BettererGoal, BettererGoalFunction, BettererTest, BettererConstraint, BettererDiff } from './types';

export type BettererOptions<TestType, SerialisedType = TestType> = {
  test: BettererTest<TestType>;
  constraint: BettererConstraint<SerialisedType>;
  goal: BettererGoal<SerialisedType>;
  diff: BettererDiff<TestType, SerialisedType>;
  isOnly?: boolean;
  isSkipped?: boolean;
};

export class Betterer<Base = unknown, Serialised = Base> {
  public constraint: BettererConstraint<Serialised>;
  public diff: BettererDiff<Base, Serialised>;
  public goal: BettererGoalFunction<Serialised>;
  public test: BettererTest<Base>;
  public isOnly = false;
  public isSkipped = false;
  public name = '';

  constructor({
    test,
    constraint,
    goal,
    diff = defaultDiff,
    isOnly = false,
    isSkipped = false
  }: BettererOptions<Base, Serialised>) {
    this.constraint = constraint;
    this.diff = diff;
    this.goal = createGoal(goal);
    this.test = test;
    this.isOnly = isOnly;
    this.isSkipped = isSkipped;
  }

  public only(): Betterer<Base, Serialised> {
    this.isOnly = true;
    return this;
  }

  public skip(): Betterer<Base, Serialised> {
    this.isSkipped = true;
    return this;
  }
}

function createGoal(goal: BettererGoal<unknown>): BettererGoalFunction<unknown> {
  if (typeof goal === 'function') {
    return goal as BettererGoalFunction<unknown>;
  }
  return (value: unknown): boolean => value === goal;
}

function defaultDiff(_: unknown, serialisedCurrent: unknown, serialisedPrevious: unknown): void {
  const diffStr =
    logDiff(serialisedPrevious, serialisedCurrent, {
      aAnnotation: 'Previous',
      bAnnotation: 'Current'
    }) || '';
  console.log(
    diffStr
      .split('\n')
      .map((line) => `  ${line}`)
      .join('\n')
  );
}
