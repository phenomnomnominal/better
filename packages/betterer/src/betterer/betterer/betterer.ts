import { BettererRun } from '../../context';
import { BettererConstraint, BettererGoal, BettererTestFunction, BettererOptions } from './types';

export class Betterer<TestType = unknown> {
  public readonly constraint: BettererConstraint<TestType>;
  public readonly goal: BettererGoal<TestType>;
  public readonly test: BettererTestFunction<TestType>;

  public readonly isBetterer = true;

  private _isOnly: boolean;
  private _isSkipped: boolean;

  constructor(options: BettererOptions<TestType>) {
    const { constraint, test, isOnly, isSkipped } = options;
    this.constraint = constraint;
    this.goal = this._createGoal(options);
    this.test = test;
    this._isOnly = isOnly || false;
    this._isSkipped = isSkipped || false;
  }

  public get isOnly(): boolean {
    return this._isOnly;
  }

  public get isSkipped(): boolean {
    return this._isSkipped;
  }

  public only(): this {
    this._isOnly = true;
    return this;
  }

  public skip(): this {
    this._isSkipped = true;
    return this;
  }

  public getExpected(run: BettererRun): TestType {
    return run.test.expected as TestType;
  }

  private _createGoal(options: BettererOptions<TestType>): BettererGoal<TestType> {
    const hasGoal = Object.hasOwnProperty.call(options, 'goal');
    if (!hasGoal) {
      return (): boolean => false;
    }
    const { goal } = options;
    if (this._isGoalFunction(goal)) {
      return goal;
    }
    return (value: unknown): boolean => value === goal;
  }

  private _isGoalFunction(goal: unknown): goal is BettererGoal<unknown> {
    return typeof goal === 'function';
  }
}

export function isBetterer(obj: unknown): obj is Betterer {
  return !!(obj as Betterer).isBetterer;
}
