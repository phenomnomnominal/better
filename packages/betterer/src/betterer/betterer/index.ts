export * from './betterer';
export * from './types';
