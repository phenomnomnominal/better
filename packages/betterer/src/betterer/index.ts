export { createBetterer } from './create-betterer';
export { Betterer, isBetterer } from './betterer';
export { BettererFiles, BettererFile, FileBetterer, createFileBetterer, isFileBetterer } from './file-betterer';
export { SerialisableBetterer, isSerialisableBetterer } from './serialisable-betterer';
export * from './betterer/types';
export * from './file-betterer/types';
export * from './serialisable-betterer/types';
export * from './types';
