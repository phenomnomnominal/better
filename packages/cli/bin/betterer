#!/usr/bin/env node
import { cli__ } from '../dist/cli.js';
cli__(process.cwd(), process.argv);
