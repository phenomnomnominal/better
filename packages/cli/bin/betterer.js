#!/usr/bin/env node
import { cliΔ } from '../dist/cli.js';
void cliΔ(globalThis.process.cwd(), globalThis.process.argv);
