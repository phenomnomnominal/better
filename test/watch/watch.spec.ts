import { betterer, BettererSuiteSummary } from '@betterer/betterer';
import assert from 'assert';

import { createFixture } from '../fixture';

describe('betterer.watch', () => {
  it('should run in watch mode', async () => {
    const { logs, paths, resolve, cleanup, writeFile } = await createFixture('watch', {
      '.betterer.ts': `
import { tsquery } from '@betterer/tsquery';

export default {
  test: () => tsquery(
    'CallExpression > PropertyAccessExpression[expression.name="console"][name.name="log"]'
  ).include('./src/**/*.ts')
};
      `
    });

    const configPaths = [paths.config];
    const resultsPath = paths.results;
    const indexPath = resolve('./src/index.ts');
    const { cwd } = paths;

    await writeFile(indexPath, `console.log('foo');console.log('foo');`);

    await betterer({ configPaths, resultsPath, cwd, workers: false });

    const suiteSummaryDefers = [
      defer<BettererSuiteSummary>(),
      defer<BettererSuiteSummary>(),
      defer<BettererSuiteSummary>()
    ];
    const [worse, same, better] = suiteSummaryDefers;

    const runner = await betterer.watch({
      configPaths,
      resultsPath,
      cwd,
      reporters: [
        '@betterer/reporter',
        {
          suiteEnd(suiteSummary: BettererSuiteSummary) {
            const suiteSummaryDefer = suiteSummaryDefers.shift();
            suiteSummaryDefer?.resolve(suiteSummary);
          }
        }
      ],
      workers: false
    });

    await writeFile(indexPath, `console.log('foo');\nconsole.log('foo');console.log('foo');`);

    const worseSuiteSummary = await worse.promise;
    const [worseRun] = worseSuiteSummary.runs;

    expect(worseRun.isWorse).toBe(true);

    await writeFile(indexPath, `console.log('foo');console.log('foo');`);

    const sameSuiteSummary = await same.promise;
    const [sameRun] = sameSuiteSummary.runs;

    expect(sameRun.isSame).toBe(true);

    await writeFile(indexPath, `console.log('bar');`);

    const betterSuiteSummary = await better.promise;
    const [betterRun] = betterSuiteSummary.runs;

    expect(betterRun.isBetter).toBe(true);

    await runner.stop();

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});

type Resolve<T> = (value: T) => void;
type Reject = (error: Error) => void;
type Defer<T> = {
  promise: Promise<T>;
  resolve: Resolve<T>;
  reject: Reject;
};

function defer<T>(): Defer<T> {
  let resolve: Resolve<T> | null = null;
  let reject: Reject | null = null;
  const promise = new Promise<T>((res, rej) => {
    resolve = res;
    reject = rej;
  });
  assert(resolve);
  assert(reject);
  return { promise, resolve, reject };
}
