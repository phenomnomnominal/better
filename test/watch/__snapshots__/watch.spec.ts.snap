// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`betterer.watch should run in watch mode 1`] = `
Array [
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms):
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
🤔 test: running \\"test\\"!
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🌟 Betterer (0ms): 1 test running...
✅ test: \\"test\\" got checked for the first time! (2 issues) 🎉
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" got checked for the first time! (2 issues) 🎉
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" got checked for the first time! (2 issues) 🎉

1 test got checked. 🤔
1 test got checked for the first time! 🎉
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Starting Betterer in watch mode! 🎉

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms):

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
🤔 test: running \\"test\\"!

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
🔥 test: \\"test\\" got worse. (1 new issue, 3 total) 😔

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
🔥 test: \\"test\\" got worse. (1 new issue, 3 total) 😔

Error: \\"test\\" got worse. (1 new issue, 3 total) 😔

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

💥 Betterer (0ms): 1 test done! 1 test errored!
🔥 test: \\"test\\" got worse. (1 new issue, 3 total) 😔
・ 2 existing issues in \\"<project>/fixtures/watch/src/index.ts\\".
・ New issue in \\"<project>/fixtures/watch/src/index.ts\\"!
・
・   <project>/fixtures/watch/src/index.ts
・   1 | console.log('foo');
・ > 2 | console.log('foo');console.log('foo');
・     |                    ^^^^^^^^^^^ TSQuery match
・

Error: \\"test\\" got worse. (1 new issue, 3 total) 😔

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checked 1 file! 🔍

・ <project>/fixtures/watch/src/index.ts

💥 Betterer (0ms): 1 test done! 1 test errored!
🔥 test: \\"test\\" got worse. (1 new issue, 3 total) 😔
・ 2 existing issues in \\"<project>/fixtures/watch/src/index.ts\\".
・ New issue in \\"<project>/fixtures/watch/src/index.ts\\"!
・
・   <project>/fixtures/watch/src/index.ts
・   1 | console.log('foo');
・ > 2 | console.log('foo');console.log('foo');
・     |                    ^^^^^^^^^^^ TSQuery match
・

Error: \\"test\\" got worse. (1 new issue, 3 total) 😔

1 test got checked. 🤔
1 test got worse. 😔


(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

💥 Betterer (0ms): 1 test done! 1 test errored!
🔥 test: \\"test\\" got worse. (1 new issue, 3 total) 😔
・ 2 existing issues in \\"<project>/fixtures/watch/src/index.ts\\".
・ New issue in \\"<project>/fixtures/watch/src/index.ts\\"!
・
・   <project>/fixtures/watch/src/index.ts
・   1 | console.log('foo');
・ > 2 | console.log('foo');console.log('foo');
・     |                    ^^^^^^^^^^^ TSQuery match
・

Error: \\"test\\" got worse. (1 new issue, 3 total) 😔

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
🤔 test: running \\"test\\"!

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
✅ test: \\"test\\" stayed the same. (2 issues) 😐

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" stayed the same. (2 issues) 😐

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checked 1 file! 🔍

・ <project>/fixtures/watch/src/index.ts

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" stayed the same. (2 issues) 😐

1 test got checked. 🤔
1 test stayed the same. 😐

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" stayed the same. (2 issues) 😐

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
🤔 test: running \\"test\\"!

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🌟 Betterer (0ms): 1 test running...
✅ test: \\"test\\" got better! (1 fixed issue, 1 remaining) 😍

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checking 1 file... 🤔

・ <project>/fixtures/watch/src/index.ts

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" got better! (1 fixed issue, 1 remaining) 😍

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Filters (press \\"f\\" to edit): No current filter patterns
Ignores (press \\"i\\" to edit): No current ignore patterns

Checked 1 file! 🔍

・ <project>/fixtures/watch/src/index.ts

🎉 Betterer (0ms): 1 test done!
✅ test: \\"test\\" got better! (1 fixed issue, 1 remaining) 😍

1 test got checked. 🤔
1 test got better! 😍

(press \\"q\\" to quit)
",
  "
   / | /     _         _   _
 '-.ooo.-'  | |__  ___| |_| |_ ___ _ __ ___ _ __
---ooooo--- | '_ // _ / __| __/ _ / '__/ _ / '__|
 .-'ooo'-.  | |_)|  __/ |_| ||  __/ | |  __/ |
   / | /    |_.__//___|/__|/__/___|_|  /___|_|

Stopping watch mode... 👋
",
]
`;
